tidb-cloud: add set up private endpoint connections (#9542)
pingcap/docs: tidb-cloud/set-up-private-endpoint-connections.md

- [AWS PrivateLink concepts](https://docs.aws.amazon.com/vpc/latest/privatelink/concepts.html)

## Restrictions

- Currently, TiDB Cloud supports private endpoint connection only when the endpoint service is hosted in AWS. If the service is hosted in Google Cloud Platform (GCP), the private endpoint is not applicable.
- The private endpoint support is provided only for the TiDB Cloud Dedicated Tier, not for the Developer Tier.
- Private endpoint connection across regions is not supported.

In most scenarios, you are recommended to use private endpoint connection over VPC peering. However, in the following scenarios, you should use VPC peering instead of private endpoint connection:

- You are using a TiCDC cluster to replicate data from a source TiDB cluster to a target TiDB cluster across regions, to get high availability. Currently, private endpoint does not support cross-region connection.
- You are using a TiCDC cluster to replicate data to a downstream cluster (such as Amazon Aurora, MySQL, and Kafka) but you cannot maintain the endpoint service on your own.
- You are connecting to PD or TiKV nodes directly.

> 

Author: done

## Set up private endpoint with AWS

> This section describes how to set up a private endpoint.

Author: ```suggestion
This section describes how to set up a private endpoint with AWS PrivateLink.
```

#### Step 2. Choose a TiDB cluster

Click the drop-down list to choose a TiDB cluster for which you want to create a private endpoint, and then click **Next**.

> **Note:**
>
> Before a cluster is created, it is not displayed in the drop-down list.

#### Step 3. Choose Service Endpoint Region

From the **Region** List, select the region in which you want to create the private endpoint. Then, click **Next**.

> **Note:**
>
> The default region is where your cluster is located. Do not change it. Cross-region private endpoint is currently not supported.

#### Step 4. Create an AWS Interface Endpoint

At this stage, TiDB Cloud begins creating an endpoint service, which takes 3 to 4 minutes. During the creation process, perform the following operations:

1. Fill in the **VPC ID** and **Subnet IDs** fields. You can get the IDs from your AWS Management Console.

    If you do not know how to get these IDs, click **Show Instruction** and you will see two snapshots of the AWS Management Console that illustrate how to get these IDs.

2. Take a note of the endpoint service name after the endpoint service creation is complete.

>     After the endpoint service is created, the endpoint service name is available in the command in the **Create VPC Interface Endpoint** area.

Reviewer: The screenshot below clearly highlights "your service name", so it is obvious.

>     ![Endpoint service name](/media/tidb-cloud/private-endpoint/private-endpoint-service-name.png)

Author: done

    2. Under **Service category**, select **Other endpoint services**.
    3. Under **Service settings**, enter the endpoint service name you have obtained from the **Interface endpoint** page of the TiDB Cloud console, and click **Verify service**.

        ![Verify endpoint service](/media/tidb-cloud/private-endpoint/create-endpoint-2.png)

    4. After the service name is verified, under **VPC**, select your VPC in the drop-down list. Then the pre-populated **Subnets** area is displayed.
    5. In the **Subnets** area, select the availabilty zones where your TiDB cluster is located. Then click **Create endpoint** at the bottom of the page.

        ![Create endpoint service 2](/media/tidb-cloud/private-endpoint/create-endpoint-3.png)

    > **Tip:**
    >
    > If your service is spanning across more than three availability zones (AZs), you might not be able to select AZs in the **Subnets** area. This issue occurs when there is an extra AZ in your selected region in addition to the AZs where your TiDB cluster is located. To solve the issue, contact [PingCAP Technical Support](https://docs.pingcap.com/tidbcloud/tidb-cloud-support).

    </div>
    <div label="AWS CLI">

>     To use the AWS CLI to create an AWS Interface Endpoint, perform the following steps:

Reviewer: not sure my comment is correct, please double check.

    1. Install AWS Command Line Interface (AWS CLI).

        ```bash
        curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
        unzip awscliv2.zip
        sudo ./aws/install
        ```

    2. Configure AWS CLI according to your account information. To get the information required by AWS CLI, see [AWS CLI configuration basics](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).

        ```bash
        aws configure
        ```

    3. Copy the command in the **Create VPC Interface Endpoint** area and run it in your terminal to create the VPC interface endpoint. Then click **Next**.

    After the endpoint service is created, the placeholders in the command are automatically replaced with the real values.

    > **Tip:**
    >

>     > - If your service is spanning across more than three availability zones (AZs), you might not be able to select AZs in the **Subnets** area. This issue occurs when there is an extra AZ in your selected region in addition to the AZs where your TiDB cluster is located. The returned error message is as follows:

Author: Yes. Users cannot select AZs in the AWS UI.
```suggestion
    > - If your service is spanning across more than three availability zones (AZs), you might not be able to select AZs in the **Subnets** area in the UI. This issue occurs when there is an extra AZ in your selected region in addition to the AZs where your TiDB cluster is located. The returned error message is as follows:
```

---
title: Set Up Private Endpoint Connections
summary: Learn how to set up private endpoint connections in TiDB Cloud.
---

# Set Up Private Endpoint Connections

> TiDB Cloud supports highy secure and one-way access to the TiDB Cloud service hosted in an AWS VPC via the [AWS PrivateLink](https://aws.amazon.com/privatelink/?privatelink-blogs.sort-by=item.additionalFields.createdDate&privatelink-blogs.sort-order=desc), as if the service were in your own VPC. A private endpoint is exposed in your VPC and you can create a connection to the TiDB Cloud service via the endpoint with permission.

Reviewer: ```suggestion
TiDB Cloud supports highly secure and one-way access to the TiDB Cloud service hosted in an AWS VPC via the [AWS PrivateLink](https://aws.amazon.com/privatelink/?privatelink-blogs.sort-by=item.additionalFields.createdDate&privatelink-blogs.sort-order=desc), as if the service were in your own VPC. A private endpoint is exposed in your VPC and you can create a connection to the TiDB Cloud service via the endpoint with permission.
```

#### Step 1. Open the private endpoint creation page

To find the entrance to creating a private endpoint, take the following steps:

1. On the TiDB Cloud console, click the **Project Settings** tab, then click **Private Endpoint** on the left menu, and the **Private Endpoint** page is displayed.
2. Click **Add** in the upper-right corner to open the creation page.

> **Tip:**
>
> Alternatively, you can find the entrance by following these steps:
>
> 1. In the TiDB Cloud console, navigate to the **Clusters** page.
> 2. Locate your cluster, and click **Connect** in the upper-right corner of the cluster area. A connection dialog box is displayed.
> 3. Select the **Private Endpoint** tab. If no private endpoint has been created, click **Create** in the dialog to open the creation page.

#### Step 2. Choose a TiDB cluster

Click the drop-down list to choose a TiDB cluster for which you want to create a private endpoint, and then click **Next**.

> **Note:**
>
> Before a cluster is created, it is not displayed in the drop-down list.

> #### Step 3. Choose Service Endpoint Region

Reviewer: ```suggestion
#### Step 3. Choose the service endpoint region
```

    ![Endpoint service name](/media/tidb-cloud/private-endpoint/private-endpoint-service-name.png)

3. Create a VPC interface endpoint in AWS. You can either use the AWS Management Console or the AWS CLI.

    <SimpleTab>
    <div label="Use AWS Console">

    To use the AWS Management Console to create the VPC interface endpoint, perform the following steps:

    1. In your AWS Management Console, go to **VPC** > **Endpoints**, and click **Create Endpoint** in the upper-right corner. The **Create endpoint** page is displayed.

        ![Create endpoint](/media/tidb-cloud/private-endpoint/create-endpoint-1.png)

    2. Under **Service category**, select **Other endpoint services**.
    3. Under **Service settings**, enter the endpoint service name you have obtained from the **Interface endpoint** page of the TiDB Cloud console, and click **Verify service**.

        ![Verify endpoint service](/media/tidb-cloud/private-endpoint/create-endpoint-2.png)

    4. After the service name is verified, under **VPC**, select your VPC in the drop-down list. Then the pre-populated **Subnets** area is displayed.

>     5. In the **Subnets** area, select the availabilty zones where your TiDB cluster is located. Then click **Create endpoint** at the bottom of the page.

Reviewer: ```suggestion
    5. In the **Subnets** area, select the availability zones where your TiDB cluster is located. Then click **Create endpoint** at the bottom of the page.
```

        ```bash
        aws configure
        ```

    3. Copy the command in the **Create VPC Interface Endpoint** area and run it in your terminal to create the VPC interface endpoint. Then click **Next**.

    After the endpoint service is created, the placeholders in the command are automatically replaced with the real values.

    > **Tip:**
    >
    > - If your service is spanning across more than three availability zones (AZs), you will get an error message indicating that the VPC endpoint service does not support the AZ of the subnet. This issue occurs when there is an extra AZ in your selected region in addition to the AZs where your TiDB cluster is located. In this case, you can contact [PingCAP Technical Support](https://docs.pingcap.com/tidbcloud/tidb-cloud-support).
    >
    > - You cannot copy the command until TiDB Cloud finishes creating endpoint service in the background.

    </div>
    </SimpleTab>

#### Step 5. Accept the endpoint connection

> Fill in the box with the your VPC endpoint ID and click **Next**.

Reviewer: ```suggestion
Fill in the box with your VPC endpoint ID and click **Next**.
```

---
title: Set Up Private Endpoint Connections
summary: Learn how to set up private endpoint connections in TiDB Cloud.
---

# Set Up Private Endpoint Connections

TiDB Cloud supports highy secure and one-way access to the TiDB Cloud service hosted in an AWS VPC via the [AWS PrivateLink](https://aws.amazon.com/privatelink/?privatelink-blogs.sort-by=item.additionalFields.createdDate&privatelink-blogs.sort-order=desc), as if the service were in your own VPC. A private endpoint is exposed in your VPC and you can create a connection to the TiDB Cloud service via the endpoint with permission.

Powered by AWS PrivateLink, the endpoint connection is secure and private, and does not expose your data to the public internet.

The architecture of the private endpoint is as follows:

![Private endpoint architecture](/media/tidb-cloud/aws-private-endpoint-arch.png)

> The cloud vendor AWS generates a specific DNS hostname for your endpoint service. Before you grant permissions to service consumers or specific AWS principals (AWS accounts, IAM users, and IAM roles), they do not have access to your endpoint service. For more detailed definitions of the private endpoint and endpoint service, see the following AWS documents:

Reviewer 2: remove DNS hostname generation.

Author: done

- [What is AWS PrivateLink?](https://docs.aws.amazon.com/vpc/latest/privatelink/what-is-privatelink.html)
- [AWS PrivateLink concepts](https://docs.aws.amazon.com/vpc/latest/privatelink/concepts.html)

The relationship of a TiDB cluster in TiDB Cloud, an endpoint service, and a private endpoint is as follows:

- Each TiDB cluster corresponds to one endpoint service, and each endpoint service corresponds to one TiDB cluster.
- Each endpoint service corresponds to one or multiple private endpoints.
- Each TiDB cluster corresponds to one or multiple private endpoints.

> **Note:**
>
> - Currently, TiDB Cloud supports private endpoint connection only when the endpoint service is hosted in AWS. If the service is hosted in Google Cloud Platform (GCP), the private endpoint is not applicable.
> - The private endpoint support is provided only for the TiDB Cloud Dedicated Tier, not for the Developer Tier.
> - Private endpoint connection across regions is not supported.

> ## Why should I use private endpoint

Reviewer 2: move to the overview of private endpoint

Author: done

- Private traffic: PrivateLink-powered endpoint traffic does not traverse the public internet. Private endpoint connection uses private IP addresses and security groups within a VPC so that the services function as if they were hosted directly within a VPC.
- Enhanced security: Compared to VPC peering that allows access to all resources, private endpoint only allows access to a specific service. In addition, private endpoint only allows one-way access, and only VPC Endpoints can initiate a connection.
- Private endpoint supports CIDR overlap.
- Simplified network management: Private endpoint is easy to use and manage because it removes the need to whitelist public IPs and manage internet connectivity with internet gateways, NAT gateways, or firewall proxies. There is no longer a need to configure an internet gateway, VPC peering connection, or Transit VPC to enable connectivity.

> **Note:**
>

> > In most scenarios, you are recommended to use private endpoint connection over VPC peering. However, in the following scenarios, you should use VPC peering instead of private endpoint connection:

Author: The restriction info is important. I prefer leaving it here.

>
> - You are using a TiCDC cluster to replicate data from a source TiDB cluster to a target TiDB cluster across regions, to get high availability. Currently, private endpoint does not support cross-region connection.
> - You are using a TiCDC cluster to replicate data to a downstream cluster (such as Amazon Aurora, MySQL, and Kafka) but you cannot maintain the endpoint service on your own.
> - You are connecting to PD or TiKV nodes directly.

## Set up private endpoint with AWS

> This section describes how to set up a private endpoint and how to connect to a private endpoint service via private endpoint.

Reviewer 2: change how to connect to a link

Author: done

#### Step 1: Find the entrance

To find the entrance to creating a private endpoint, take one of the following methods:

- From the **Project Settings** page:

    1. On the TiDB Cloud console, click the **Project Settings** tab, then click **Private Endpoint** on the left menu, and the **Private Endpoint** page is displayed. If there is any existing private endpoint, it is displayed on this page.
    2. Click **Add** in the upper-right corner to open the creation page.

- From the **Connect to TiDB** dialog of an active cluster:

    1. In the TiDB Cloud console, navigate to the **Active Clusters** page and click the name of your newly created cluster.
    2. Click **Connect**. The **Connect to TiDB** dialog box is displayed.
    3. Select the **Private Endpoint** tab. If no private endpoint has been created, click **Create** on the dialog to open the creation page.

On the creation page, a flow bar is displayed indicating the stages of creating a private endpoint: **Choose Cluster** > **Service Endpoint** > **Interface Endpoint** > **Accept Endpoint Connection** > **Enable Private DNS**.

#### Step 2: Choose a TiDB cluster

> After you open the creation page for a private endpoint, you are at the **Choose Cluster** stage. Click the drop-down list to choose a TiDB cluster for which you want to create a private endpoint, and then click **Next**.

Reviewer 2: remove stage

Author: done

    After the endpoint service creation is complete, the endpoint service name and the **Next** button are available, and the placeholders in the command in the **Create VPC Interface Endpoint** area are automatically replaced with the real values.

2. Then you need to create the VPC interface endpoint in AWS. You can either use the AWS Management Console or the AWS CLI.

    <SimpleTab>
    <div label="Use AWS Console">

    To use the AWS Management Console to create the VPC interface, perform the following steps:

    1. In your AWS Management Console, go to **VPC** > **Endpoint**, and click **Create Endpoint** in the upper-right corner. The **Create endpoint** page is displayed.
    2. Under **Service category**, select **Other endpoint services**.
    3. Under **Service settings**, enter the endpoint service name you have obtained from the **Interface endpoint** page of the TiDB Cloud console, and click **Verify service**.
    4. After the service name is verified, under **VPC**, select your VPC in the drop-down list. Then the pre-populated **Subnets** area is displayed.

>     5. In the **Subnets** area, select the availabilty zones where your TiDB cluster is located. Then click **Create endpoint**.

Reviewer 2: need pic

        ```

    2. Configure AWS CLI according to your account information. To get the information required by AWS CLI, see [AWS CLI configuration basics](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).

        ```bash
        aws configure
        ```

    3. Copy the command in the **Create VPC Interface Endpoint** area and run it in your terminal to create the VPC interface endpoint.

    > **Tip:**
    >
    > - If your service is spanning across more than three availability zones (AZs), an error is returned. To resolve the issue, contact [PingCAP Technical Support](https://docs.pingcap.com/tidbcloud/tidb-cloud-support).
    >
    >     In addition to the AZs where your TiDB cluster is located, if there is an extra AZ in your selected region, this issue will occur.
    >
    > - You cannot copy the command until TiDB Cloud finishes creating endpoint service in the background.

    </div>
    </SimpleTab>

    Then click **Next**.

#### Step 5: Accept the endpoint connection

After you have created an endpoint service, you are at the **Accept Endpoint Connection** stage. Fill in the box with the your VPC endpoint ID and click **Next**.

> You can get your VPC endpoint ID from your AWS Management Console. If you do not know how to get it, click **Show Instruction** and you will see a snapshot of the AWS Management Console that illustrates how to get it. To fold the snapshot, click the **Hide instruction**.

Reviewer 2: remove instruction op

Author: done

#### Step 6: Enable Private DNS

After you have accepted the endpoint connection, you are at the **Enable Private DNS** stage. Click the **Copy** button to copy the command and run it in your AWS CLI. The `<your_vpc_endpoint_id>` placeholder is automatically replaced with the value you have provided in Step 5.

Then click **Create** to finalize the creation of the private endpoint.

> ### Connect to a private endpoint service

Reviewer 2: leave a link

Author: done

Take the following steps to connect to a private endpoint service using a private endpoint:

1. In the TiDB Cloud console, navigate to the **Active Clusters** page and click the name of your newly created cluster.
2. Click **Connect**. The **Connect to TiDB** dialog box is displayed.
3. Select the **Private Endpoint** tab. If you have created a private endpoint, it is displayed under **Step 1: Create Private Endpoint**.
4. Under **Step 2: Connect your application**, click the tab of your preferred connection method, and then connect to your cluster with the connection string. The placeholders `<cluster_endpoint_name>:<port>` in the connection string are automatically replaced with the real values.

> ## Billing information

Reviewer 2: remove billing info to https://docs.pingcap.com/tidbcloud/tidb-cloud-billing#data-transfer-cost

Author: done

Powered by AWS PrivateLink, the endpoint connection is secure and private, and does not expose your data to the public internet.

The architecture of the private endpoint is as follows:

![Private endpoint architecture](/media/tidb-cloud/aws-private-endpoint-arch.png)

The cloud vendor AWS generates a specific DNS hostname for your endpoint service. Before you grant permissions to service consumers or specific AWS principals (AWS accounts, IAM users, and IAM roles), they do not have access to your endpoint service. For more detailed definitions of the private endpoint and endpoint service, see the following AWS documents:

- [What is AWS PrivateLink?](https://docs.aws.amazon.com/vpc/latest/privatelink/what-is-privatelink.html)
- [AWS PrivateLink concepts](https://docs.aws.amazon.com/vpc/latest/privatelink/concepts.html)

> The relationship of a TiDB cluster in TiDB Cloud, an endpoint service, and a private endpoint is as follows:

Reviewer 2: it's difficult to understand, change another way

> **Note:**
>
> In most scenarios, you are recommended to use private endpoint connection over VPC peering. However, in the following scenarios, you should use VPC peering instead of private endpoint connection:
>
> - You are using a TiCDC cluster to replicate data from a source TiDB cluster to a target TiDB cluster across regions, to get high availability. Currently, private endpoint does not support cross-region connection.
> - You are using a TiCDC cluster to replicate data to a downstream cluster (such as Amazon Aurora, MySQL, and Kafka) but you cannot maintain the endpoint service on your own.
> - You are connecting to PD or TiKV nodes directly.

## How to use private endpoint

This section describes how to create, edit, delete or terminate a private endpoint, and how to connect to a private endpoint service via private endpoint.

### Create a private endpoint

> Before you create a private endpoint, make sure that you are using TiDB Cloud Dedicated Tier.

Reviewer 2: ```suggestion
```
remove it, it's mention in Prerequisites

    3. Select the **Private Endpoint** tab. If no private endpoint has been created, click **Create** on the dialog to open the creation page.

On the creation page, a flow bar is displayed indicating the stages of creating a private endpoint: **Choose Cluster** > **Service Endpoint** > **Interface Endpoint** > **Accept Endpoint Connection** > **Enable Private DNS**.

#### Step 2. Choose a TiDB cluster

After you open the creation page for a private endpoint, you are at the **Choose Cluster** stage. Click the drop-down list to choose a TiDB cluster for which you want to create a private endpoint, and then click **Next**.

> **Note:**
>
> - Only the Dedicated Tier clusters are displayed in the drop-down list. Developer Tier clusters are not displayed.
> - If a cluster is being created, it is not displayed in the drop-down list.
> - If there is no existing created cluster, you will see **Create Cluster** on the right side of the drop-down list. Click **Create Cluter** to create a TiDB cluster. For detailed instructions, see [Create a TiDB Cluster in TiDB Cloud](/tidb-cloud/create-tidb-cluster.md).

#### Step 3. Choose Service Endpoint Region

> After you have choosen a TiDB cluster, you are at the **Service Endpoint** stage. Click the drop-down list to choose the region where your cluster is located, and then click **Next**.

Reviewer 2: ```suggestion
From the **Region** List, select the region in which you want to create the private endpoint. Click **Next**.
```

> Note that you cannot choose other regions. Cross-region private endpoint is currently not supported.

Reviewer 2: ```suggestion
> **Note:**
> 
> - The default value of the region is where your cluster is located, please don't change it. Cross-region private endpoint is currently not supported.```

    </div>
    <div label="AWS CLI">

    To use the AWS CLI to create an AWS Interface Endpoint, perform the following steps:

    1. Install AWS Command Line Interface (AWS CLI).

        ```bash
        curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
        unzip awscliv2.zip
        sudo ./aws/install
        ```

    2. Configure AWS CLI according to your account information. To get the information required by AWS CLI, see [AWS CLI configuration basics](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).

        ```bash
        aws configure
        ```

    3. Replace the following variable values with your account information.

        ```bash
        # Sets up the related variables.
        app_region="<APP Region>"
        app_vpc_id="<Your VPC ID>"
        tidbcloud_project_cidr="<TiDB Cloud Project VPC CIDR>"
        ```

        For example:

        ```
        # Sets up the related variables
        pcx_tidb_to_app_id="pcx-069f41efddcff66c8"
        app_region="us-west-2"
        app_vpc_id="vpc-0039fb90bb5cf8698"
        tidbcloud_project_cidr="10.250.0.0/16"
        ```

>     4. Copy the command in the **Create VPC Interface Endpoint** area and run it in your terminal to create the VPC interface endpoint.

Reviewer 2: ```suggestion
    3. Copy the command in the **Create VPC Interface Endpoint** area and run it in your terminal to create the VPC interface endpoint.
```

- From the **Connect to TiDB** dialog of an active cluster:

    1. In the TiDB Cloud console, navigate to the **Active Clusters** page and click the name of your newly created cluster.
    2. Click **Connect**. The **Connect to TiDB** dialog box is displayed.
    3. Select the **Private Endpoint** tab. If no private endpoint has been created, click **Create** on the dialog to open the creation page.

On the creation page, a flow bar is displayed indicating the stages of creating a private endpoint: **Choose Cluster** > **Service Endpoint** > **Interface Endpoint** > **Accept Endpoint Connection** > **Enable Private DNS**.

#### Step 2. Choose a TiDB cluster

After you open the creation page for a private endpoint, you are at the **Choose Cluster** stage. Click the drop-down list to choose a TiDB cluster for which you want to create a private endpoint, and then click **Next**.

> **Note:**
>
> - Before the cluster is created,  You can't create a private endpoint for the cluster.

#### Step 3. Choose Service Endpoint Region

From the **Region** List, select the region in which you want to create the private endpoint. Click **Next**.

> **Note:**
> 

> > - The default value of the region is where your cluster is located, please don't change it. Cross-region private endpoint is currently not supported.```

Reviewer 2: ```suggestion
> - The default value of the region is where your cluster is located, please don't change it. Cross-region private endpoint is currently not supported.
```

    5. In the **Subnets** area, select the availabilty zones where your TiDB cluster is located. Then click **Create endpoint**.

    > **Tip:**
    >
    > If your service is spanning across more than three availability zones (AZs), you might not be able to select AZs in the **Subnets** area. To solve the issue, contact [PingCAP Technical Support](https://docs.pingcap.com/tidbcloud/tidb-cloud-support).
    >
    > In addition to the AZs where your TiDB cluster is located, if there is an extra AZ in your selected region, this issue will occur.

    </div>
    <div label="AWS CLI">

    To use the AWS CLI to create an AWS Interface Endpoint, perform the following steps:

    1. Install AWS Command Line Interface (AWS CLI).

        ```bash
        curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
        unzip awscliv2.zip
        sudo ./aws/install
        ```

    2. Configure AWS CLI according to your account information. To get the information required by AWS CLI, see [AWS CLI configuration basics](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).

        ```bash
        aws configure
        ```

>     3. Replace the following variable values with your account information.

Reviewer 2: ```suggestion
```

>         ```bash

Reviewer 2: ```suggestion
```

>         # Sets up the related variables.

Reviewer 2: ```suggestion
```

>         app_region="<APP Region>"

Reviewer 2: ```suggestion
```

>         app_vpc_id="<Your VPC ID>"

Reviewer 2: ```suggestion
```

>         tidbcloud_project_cidr="<TiDB Cloud Project VPC CIDR>"

Reviewer 2: ```suggestion
```

        ```

>         For example:

Reviewer 2: ```suggestion
```

>         ```

Reviewer 2: ```suggestion
```

        # Sets up the related variables

>         pcx_tidb_to_app_id="pcx-069f41efddcff66c8"

Reviewer 2: ```suggestion
```

>         app_region="us-west-2"

Reviewer 2: ```suggestion
```

>         app_vpc_id="vpc-0039fb90bb5cf8698"

Reviewer 2: ```suggestion
```

>         tidbcloud_project_cidr="10.250.0.0/16"

Reviewer 2: ```suggestion
```

>         ```

Reviewer 2: ```suggestion
```

    4. Copy the command in the **Create VPC Interface Endpoint** area and run it in your terminal to create the VPC interface endpoint.

    > **Tip:**

>     >

Reviewer 2: ```suggestion
    > 
    > You can't copy the command until TiDB Cloud finishes creating endpoint service in the background.
```

4. Under **Step 2: Connect your application**, click the tab of your preferred connection method, and then connect to your cluster with the connection string. The placeholders `<cluster_endpoint_name>:<port>` in the connection string are automatically replaced with the real values.

## Billing information

The private endpoint is available only for Dedicated Tier clusters.

The cost of private endpoint contains the following parts:

- The cost of the private endpoint: $0.01/hour per VPC per endpoint. For details, see [AWS PrivateLink pricing - Interface Endpoint pricing](https://aws.amazon.com/privatelink/pricing/?nc1=h_ls).
- The cost of endpoint service or network load balancer. For details, see [AWS Elastic Load Balancing pricing](https://aws.amazon.com/elasticloadbalancing/pricing/).
- The cost of data transfer, which is the same as the EC2 data transfer charges. For details, see [TiDB Cloud - Data transfer cost](https://docs.pingcap.com/tidbcloud/tidb-cloud-billing#data-transfer-cost) or [AWS EC2 pricing](https://aws.amazon.com/ec2/pricing/on-demand/).

> ## Status information of private endpoint and private endpoint service

Author: I have moved this part to TiDB Cloud FAQ.

> You can see the status of a private endpoint or private endpoint service on the [**Private Endpoint** page](#step-1-find-the-entrance).

Reviewer 2: ```suggestion
```
